app/services/katello/pulp3/repository/generic.rb

```diff
@@ -7,11 +7,16 @@ def copy_content_for_source(source_repository, _options = {})
         end
 
         def distribution_options(path)
-          {
+          options = {
             base_path: path,
-            publication: repo.publication_href,
             name: "#{generate_backend_object_name}"
           }
+
+          unless ::Katello::RepositoryTypeManager.find(repo.content_type).pulp3_skip_publication
```

Comment:
This is needed to support repository types that use the `auto-publish` pulpcore feature.